### ISSUE TITLE
explanation of the implicit method

### ISSUE THREAD

**Issue by runjerry** [google/neural-tangents#2]
Hi,

I wonder if you might possibly provide a description of the implicit method used in the _compute_ntk() function or point me to some reference. I find the current codes concise but hard to follow. Thanks!

Regards,
Jerry

**Comment by Sohl-Dickstein** [comment]
Sorry for the very long lag on the response to your question!!

The implicit kernel is derived by observing that:
    Theta = J(X_1) J(X_2)^T = d[J(X_1) J(X_2)^T v] / d[v^T],
for a vector v. This allows the computation of the NTK to be phrased as: a(v) = J(X_2)^T v`, which is computed by a vector-Jacobian product; b(v) = J(X_1) a(v) which is computed by a Jacobian-vector product; and Theta = d[b(v)] / d[v^T] which is computed by taking the Jacobian of b(v).

I also just edited the docstring for that function to clarify this. The description above should also appear in the function header within the next few days.

Thanks for your interest and careful reading of the code!

**Comment by romanngg** [comment]
65215c5e75b4a9a471706e211e1f0b1040c0aff1
